homeworkWeekend9/src/ba/bitcamp/homewirkWeekend9/HTTPImageServer/HTTPServer.java

```
@@ -0,0 +1,148 @@
+
+						/*
+						 * If initial message is "/" or "/favicon.ico" that
+						 * means the client is browser client and server creates
+						 * web site with links for pictures from base.
+						 */
+					} else if (initialMessageParts[1].equals("/") || initialMessageParts[1].equals("/favicon.ico")) {
+
+						// Reading HTML code from index.html file
+						BufferedReader readerHtml = new BufferedReader(new FileReader(new File("src/index.html")));
+
+						String htmlText = "";
+						while (readerHtml.ready()) {
+							htmlText += readerHtml.readLine();
+						}
+
+						BufferedWriter writeHtml = new BufferedWriter(new OutputStreamWriter(client.getOutputStream()));
+
+						// Writing HTML code
+						writeHtml.write(htmlText);
+						writeHtml.newLine();
+
+						// Adding links for each picture from list
+						for (File picture : listOfFiles) {
+							String pictureName = picture.getName().split("\\.")[0];
+							writeHtml.write("<a href=/" + pictureName + ">" + pictureName + "<br>");
+							writeHtml.newLine();
+							writeHtml.flush();
+						}
+
+						writeHtml.write("</body></html>");
+						writeHtml.close();
+						readerHtml.close();
+
+						/*
+						 * Opening link from picture that browser client chooses
+						 * and creating new web page with that picture on it
+						 */
+					} else {
+
+						String pictureName = initialMessageParts[1].substring(1) + ".jpg";
+
+						OutputStream outputStream = client.getOutputStream();
+						FileInputStream fileInputStream = new FileInputStream("images/" + pictureName);
+
+						byte[] buffer = new byte[1024];
+						int readBytes;
+
+						// Sending certain picture to her local web page
+						while ((readBytes = fileInputStream.read(buffer, 0, buffer.length)) > 0) {
+							outputStream.write(buffer, 0, readBytes);
+							outputStream.flush();
+						}
+
+						fileInputStream.close();
+						outputStream.close();
+
+					}
+				} catch (NullPointerException e) {
+				}
```

Empty catch blocks are really dangerous.